### PR TITLE
fix(agent): ensure package version retrieval does not error out

### PR DESCRIPTION
**Pull request by hahuja2**
This PR does the following:

- For Drupal the `Drupal::VERSION` class constant is retrieved using zend calls from C and does not require using `zend_eval_string`.  The retrieval is robust against the this constant not being defined.
- For Wordpress, the presence of the `wp_version` value in `$GLOBAL` is verified  before trying to retrieve the value.

### CODE REVIEW

**Review by bduranleau-nr — APPROVED** [review]

File: agent/fw_drupal8.c [hunk 1]
```diff
@@ -534,19 +534,29 @@ NR_PHP_WRAPPER(nr_drupal8_name_the_wt_via_symfony) {
...
-  zval retval;
-  int result
-      = zend_eval_string(string, &retval, "Retrieve Drupal Version");
-  
+  zval* zval_version = NULL;
+  zend_class_entry* class_entry = NULL;
+
+  class_entry = nr_php_find_class("drupal");
+  if (NULL == class_entry) {
+    nrl_verbosedebug(NRL_INSTRUMENT, "%s: Drupal class not found", __func__);
```
Comment (lavarou) [inline agent/fw_drupal8.c:542]: Not any Drupal class, the 'Drupal' class:
```suggestion
    nrl_verbosedebug(NRL_INSTRUMENT, "%s: 'Drupal' class not found", __func__);
```

File: agent/fw_drupal8.c [hunk 2]
```diff
@@ -534,19 +534,29 @@ NR_PHP_WRAPPER(nr_drupal8_name_the_wt_via_symfony) {
 NR_PHP_WRAPPER_END
 
 void nr_drupal_version() {
-  char* string = "Drupal::VERSION;";
-  zval retval;
-  int result
-      = zend_eval_string(string, &retval, "Retrieve Drupal Version");
-  
+  zval* zval_version = NULL;
+  zend_class_entry* class_entry = NULL;
```
Comment (lavarou) [inline agent/fw_drupal8.c:538]: Maybe add a comment explaining that all this function does is read `Drupal::VERSION` constant in a safe way?

File: agent/fw_drupal8.c [hunk 3]
```diff
@@ -534,19 +534,29 @@ NR_PHP_WRAPPER(nr_drupal8_name_the_wt_via_symfony) {
...
+  }
+
   // Add php package to transaction
-  if (result == SUCCESS) {
-    if (Z_TYPE(retval) == IS_STRING) {
-      char* version = Z_STRVAL(retval);
-      nr_txn_add_php_package(NRPRG(txn), "drupal/core", version);
-    }
-    zval_dtor(&retval);
+  if (IS_STRING == Z_TYPE_P(zval_version)) {
```
Comment (lavarou) [inline agent/fw_drupal8.c:554]: Consider using a robust `nr_php_is_zval_valid_string` to check if `zval_version` is valid:
```suggestion
  if (nr_php_is_zval_valid_string(zval_version)) {
```

File: agent/fw_wordpress.c [hunk 4]
```diff
@@ -658,14 +658,27 @@ NR_PHP_WRAPPER_END
...
-  int result = zend_eval_string(string, &retval,
-                                "Retrieve Wordpress Version");
-  
+  int result
+      = zend_eval_string(func_string, &retval, "Get Wordpress Version");
   // Add php package to transaction
-  if (result == SUCCESS) {
-    if (Z_TYPE(retval) == IS_STRING) {
+  if (SUCCESS == result) {
+    if (IS_STRING == Z_TYPE(retval)) {
```
Comment (lavarou) [inline agent/fw_wordpress.c:681]: Consider using a robust `nr_php_is_zval_valid_string` to check if `retval` is valid:
```suggestion
    if (nr_php_is_zval_valid_string(&retval)) {
```

**Review by lavarou — APPROVED** [review]
Don't we want to keep this debug log in: https://github.com/newrelic/newrelic-php-agent/pull/833/commits/fe69cbedecca950bef3b00681865a33da098e49b?

File: agent/fw_drupal8.c (hunk 2 above)
Comment (mfulb) [reply agent/fw_drupal8.c:546]: Good idea - done with [b3bc447](https://github.com/newrelic/newrelic-php-agent/pull/833/commits/b3bc4476f8f69b924da3b874dc7b5eacfb39a95b) .

File: agent/fw_drupal8.c [hunk 5]
```diff
@@ -533,20 +533,38 @@ NR_PHP_WRAPPER(nr_drupal8_name_the_wt_via_symfony) {
 }
 NR_PHP_WRAPPER_END
 
+/*
+ * Drupal stores the version of the framework in the class constant
+ * Drupal::VERSION. This code first verifies the 'Drupal' class exists (note
+ * having to pass the lower case name of the class). If present then an attempt
+ * is made to retrieve the 'VERSION' class constant. Both of these checks rely
+ * on existing "nr_" routines that have been designed to be robust and will not
+ * cause an issue if either check were to fail.
```
Comment (lavarou) [inline agent/fw_drupal8.c:542]:
```suggestion
 * cause an issue in user's application if either check were to fail.
```

**Comment by hahuja2** [comment]
> Don't we want to keep this debug log in: [fe69cbe](https://github.com/newrelic/newrelic-php-agent/commit/fe69cbedecca950bef3b00681865a33da098e49b)?

This was debugging code that we determined is not necessary and have decided to keep it out for now.

**Pull request merged**